### DISCUSSION TITLE
Assistance in installing cx_Freeze for Win11?

### DISCUSSION THREAD

**Discussion by ken-at-em** [marcelotduarte/cx_Freeze#1743] [Q&A] [answered]
Hi There,

I am trying to install cx_Freeze (Win11, Python 3.11.1, venv)

Here's the error that I am getting. Any guidance would be most appreciated.

pip install --upgrade cx_Freeze
Collecting cx_Freeze
  Using cached cx_Freeze-6.13.1.tar.gz (1.5 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting packaging>=21.0
  Using cached packaging-23.0-py3-none-any.whl (42 kB)
Requirement already satisfied: setuptools<66,>=59.0.1 in c:\users\kenneth.miller\work\python\mcm\venv\lib\site-packages (from cx_Freeze) (65.5.0)
Collecting lief>=0.12.0
  Using cached lief-0.12.3-cp311-cp311-win_amd64.whl (5.0 MB)
Building wheels for collected packages: cx_Freeze
  Building wheel for cx_Freeze (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for cx_Freeze (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [111 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-311
      creating build\lib.win-amd64-cpython-311\cx_Freeze
      copying cx_Freeze\cli.py -> build\lib.win-amd64-cpython-311\cx_Freeze
      copying cx_Freeze\common.py -> build\lib.win-amd64-cpython-311\cx_Freeze
      copying cx_Freeze\darwintools.py -> build\lib.win-amd64-cpython-311\cx_Freeze
      copying cx_Freeze\dist.py -> build\lib.win-amd64-cpython-311\cx_Freeze
      copying cx_Freeze\exception.py -> build\lib.win-amd64-cpython-311\cx_Freeze
      copying cx_Freeze\executable.py -> build\lib.win-amd64-cpython-311\cx_Freeze
      copying cx_Freeze\finder.py -> build\lib.win-amd64-cpython-311\cx_Freeze
      copying cx_Freeze\freezer.py -> build\lib.win-amd64-cpython-311\cx_Freeze
      copying cx_Freeze\module.py -> build\lib.win-amd64-cpython-311\cx_Freeze
      copying cx_Freeze\parser.py -> build\lib.win-amd64-cpython-311\cx_Freeze
      copying cx_Freeze\sandbox.py -> build\lib.win-amd64-cpython-311\cx_Freeze
      copying cx_Freeze\setupwriter.py -> build\lib.win-amd64-cpython-311\cx_Freeze
      copying cx_Freeze\winmsvcr.py -> build\lib.win-amd64-cpython-311\cx_Freeze
      copying cx_Freeze\winversioninfo.py -> build\lib.win-amd64-cpython-311\cx_Freeze
      copying cx_Freeze\_compat.py -> build\lib.win-amd64-cpython-311\cx_Freeze
      copying cx_Freeze\__init__.py -> build\lib.win-amd64-cpython-311\cx_Freeze
      copying cx_Freeze\__main__.py -> build\lib.win-amd64-cpython-311\cx_Freeze
      creating build\lib.win-amd64-cpython-311\cx_Freeze\bases
      copying cx_Freeze\bases\__init__.py -> build\lib.win-amd64-cpython-311\cx_Freeze\bases
      creating build\lib.win-amd64-cpython-311\cx_Freeze\command
      copying cx_Freeze\command\bdist_mac.py -> build\lib.win-amd64-cpython-311\cx_Freeze\command
      copying cx_Freeze\command\bdist_msi.py -> build\lib.win-amd64-cpython-311\cx_Freeze\command
      copying cx_Freeze\command\bdist_rpm.py -> build\lib.win-amd64-cpython-311\cx_Freeze\command
      copying cx_Freeze\command\build.py -> build\lib.win-amd64-cpython-311\cx_Freeze\command
      copying cx_Freeze\command\build_exe.py -> build\lib.win-amd64-cpython-311\cx_Freeze\command
      copying cx_Freeze\command\install.py -> build\lib.win-amd64-cpython-311\cx_Freeze\command
      copying cx_Freeze\command\install_exe.py -> build\lib.win-amd64-cpython-311\cx_Freeze\command
      copying cx_Freeze\command\_bdist_msi.py -> build\lib.win-amd64-cpython-311\cx_Freeze\command
      copying cx_Freeze\command\_pydialog.py -> build\lib.win-amd64-cpython-311\cx_Freeze\command
      copying cx_Freeze\command\__init__.py -> build\lib.win-amd64-cpython-311\cx_Freeze\command
      creating build\lib.win-amd64-cpython-311\cx_Freeze\hooks
      copying cx_Freeze\hooks\crypto.py -> build\lib.win-amd64-cpython-311\cx_Freeze\hooks
      copying cx_Freeze\hooks\cv2.py -> build\lib.win-amd64-cpython-311\cx_Freeze\hooks
      copying cx_Freeze\hooks\exclude.py -> build\lib.win-amd64-cpython-311\cx_Freeze\hooks
      copying cx_Freeze\hooks\numpy.py -> build\lib.win-amd64-cpython-311\cx_Freeze\hooks
      copying cx_Freeze\hooks\scipy.py -> build\lib.win-amd64-cpython-311\cx_Freeze\hooks
      copying cx_Freeze\hooks\_qthooks.py -> build\lib.win-amd64-cpython-311\cx_Freeze\hooks
      copying cx_Freeze\hooks\__init__.py -> build\lib.win-amd64-cpython-311\cx_Freeze\hooks
      creating build\lib.win-amd64-cpython-311\cx_Freeze\initscripts
      copying cx_Freeze\initscripts\console.py -> build\lib.win-amd64-cpython-311\cx_Freeze\initscripts
      copying cx_Freeze\initscripts\consolesetlibpath.py -> build\lib.win-amd64-cpython-311\cx_Freeze\initscripts
      copying cx_Freeze\initscripts\sharedlib.py -> build\lib.win-amd64-cpython-311\cx_Freeze\initscripts
      copying cx_Freeze\initscripts\sharedlibsource.py -> build\lib.win-amd64-cpython-311\cx_Freeze\initscripts
      copying cx_Freeze\initscripts\__init__.py -> build\lib.win-amd64-cpython-311\cx_Freeze\initscripts
      copying cx_Freeze\initscripts\__startup__.py -> build\lib.win-amd64-cpython-311\cx_Freeze\initscripts
      creating build\lib.win-amd64-cpython-311\cx_Freeze\hooks\pyqt5
      copying cx_Freeze\hooks\pyqt5\add_library.py -> build\lib.win-amd64-cpython-311\cx_Freeze\hooks\pyqt5
      copying cx_Freeze\hooks\pyqt5\debug.py -> build\lib.win-amd64-cpython-311\cx_Freeze\hooks\pyqt5
      copying cx_Freeze\hooks\pyqt5\__init__.py -> build\lib.win-amd64-cpython-311\cx_Freeze\hooks\pyqt5
      creating build\lib.win-amd64-cpython-311\cx_Freeze\hooks\pyqt6
      copying cx_Freeze\hooks\pyqt6\debug.py -> build\lib.win-amd64-cpython-311\cx_Freeze\hooks\pyqt6
      copying cx_Freeze\hooks\pyqt6\__init__.py -> build\lib.win-amd64-cpython-311\cx_Freeze\hooks\pyqt6
      creating build\lib.win-amd64-cpython-311\cx_Freeze\hooks\pyside2
      copying cx_Freeze\hooks\pyside2\debug.py -> build\lib.win-amd64-cpython-311\cx_Freeze\hooks\pyside2
      copying cx_Freeze\hooks\pyside2\resource.py -> build\lib.win-amd64-cpython-311\cx_Freeze\hooks\pyside2
      copying cx_Freeze\hooks\pyside2\__init__.py -> build\lib.win-amd64-cpython-311\cx_Freeze\hooks\pyside2
      creating build\lib.win-amd64-cpython-311\cx_Freeze\hooks\pyside6
      copying cx_Freeze\hooks\pyside6\debug.py -> build\lib.win-amd64-cpython-311\cx_Freeze\hooks\pyside6
      copying cx_Freeze\hooks\pyside6\resource.py -> build\lib.win-amd64-cpython-311\cx_Freeze\hooks\pyside6
      copying cx_Freeze\hooks\pyside6\__init__.py -> build\lib.win-amd64-cpython-311\cx_Freeze\hooks\pyside6
      running egg_info
      writing cx_Freeze.egg-info\PKG-INFO
      writing dependency_links to cx_Freeze.egg-info\dependency_links.txt
      writing entry points to cx_Freeze.egg-info\entry_points.txt
      writing requirements to cx_Freeze.egg-info\requires.txt
      writing top-level names to cx_Freeze.egg-info\top_level.txt
      reading manifest file 'cx_Freeze.egg-info\SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      adding license file 'doc/src/license.rst'
      writing manifest file 'cx_Freeze.egg-info\SOURCES.txt'
      copying cx_Freeze\hooks\pyside2\qt.conf -> build\lib.win-amd64-cpython-311\cx_Freeze\hooks\pyside2
      copying cx_Freeze\hooks\pyside2\resource.qrc -> build\lib.win-amd64-cpython-311\cx_Freeze\hooks\pyside2
      copying cx_Freeze\hooks\pyside2\resource.sh -> build\lib.win-amd64-cpython-311\cx_Freeze\hooks\pyside2
      copying cx_Freeze\hooks\pyside6\qt.conf -> build\lib.win-amd64-cpython-311\cx_Freeze\hooks\pyside6
      copying cx_Freeze\hooks\pyside6\resource.qrc -> build\lib.win-amd64-cpython-311\cx_Freeze\hooks\pyside6
      copying cx_Freeze\hooks\pyside6\resource.sh -> build\lib.win-amd64-cpython-311\cx_Freeze\hooks\pyside6
      running build_ext
      creating build\temp.win-amd64-cpython-311
      creating build\temp.win-amd64-cpython-311\Release
      creating build\temp.win-amd64-cpython-311\Release\source
      creating build\temp.win-amd64-cpython-311\Release\source\bases
      "C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\bin\HostX86\x64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -IC:\Users\Kenneth.Miller\Work\python\mcm\venv\include -IC:\Users\Kenneth.Miller\AppData\Local\Programs\Python\Python311\include -IC:\Users\Kenneth.Miller\AppData\Local\Programs\Python\Python311\Include "-IC:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include" "-IC:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\ATLMFC\include" "-IC:\Program Files\Microsoft Visual Studio\2022\Community\VC\Auxiliary\VS\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\um" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\shared" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\winrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\cppwinrt" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um" /Tcsource/bases/console.c /Fobuild\temp.win-amd64-cpython-311\Release\source/bases/console.obj
      console.c
      C:\Users\Kenneth.Miller\AppData\Local\Temp\pip-install-nd7imgba\cx-freeze_fb866067fc8f42a9bec23e8ae76289f0\source\bases\common.c(170): warning C4996: 'Py_SetProgramName': deprecated in 3.11
      C:\Users\Kenneth.Miller\AppData\Local\Temp\pip-install-nd7imgba\cx-freeze_fb866067fc8f42a9bec23e8ae76289f0\source\bases\common.c(171): warning C4996: 'Py_SetPath': deprecated in 3.11
      C:\Users\Kenneth.Miller\AppData\Local\Temp\pip-install-nd7imgba\cx-freeze_fb866067fc8f42a9bec23e8ae76289f0\source\bases\common.c(173): warning C4996: 'PySys_SetArgvEx': deprecated in 3.11
      "C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\bin\HostX86\x64\link.exe" /nologo /INCREMENTAL:NO /LTCG /MANIFEST:EMBED,ID=1 /LIBPATH:C:\Users\Kenneth.Miller\Work\python\mcm\venv\libs /LIBPATH:C:\Users\Kenneth.Miller\AppData\Local\Programs\Python\Python311\libs /LIBPATH:C:\Users\Kenneth.Miller\AppData\Local\Programs\Python\Python311 /LIBPATH:C:\Users\Kenneth.Miller\Work\python\mcm\venv\PCbuild\amd64 "/LIBPATH:C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\ATLMFC\lib\x64" "/LIBPATH:C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\lib\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\lib\um\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.22000.0\ucrt\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\\lib\10.0.22000.0\\um\x64" build\temp.win-amd64-cpython-311\Release\source/bases/console.obj /OUT:build\lib.win-amd64-cpython-311\cx_Freeze\bases\console-cp311-win_amd64.exe /MANIFEST
      Generating code
      Finished generating code
      "C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\bin\HostX86\x64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -IC:\Users\Kenneth.Miller\Work\python\mcm\venv\include -IC:\Users\Kenneth.Miller\AppData\Local\Programs\Python\Python311\include -IC:\Users\Kenneth.Miller\AppData\Local\Programs\Python\Python311\Include "-IC:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include" "-IC:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\ATLMFC\include" "-IC:\Program Files\Microsoft Visual Studio\2022\Community\VC\Auxiliary\VS\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\um" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\shared" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\winrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\cppwinrt" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um" /Tcsource/bases/Win32GUI.c /Fobuild\temp.win-amd64-cpython-311\Release\source/bases/Win32GUI.obj
      Win32GUI.c
      C:\Users\Kenneth.Miller\AppData\Local\Temp\pip-install-nd7imgba\cx-freeze_fb866067fc8f42a9bec23e8ae76289f0\source\bases\common.c(170): warning C4996: 'Py_SetProgramName': deprecated in 3.11
      C:\Users\Kenneth.Miller\AppData\Local\Temp\pip-install-nd7imgba\cx-freeze_fb866067fc8f42a9bec23e8ae76289f0\source\bases\common.c(171): warning C4996: 'Py_SetPath': deprecated in 3.11
      C:\Users\Kenneth.Miller\AppData\Local\Temp\pip-install-nd7imgba\cx-freeze_fb866067fc8f42a9bec23e8ae76289f0\source\bases\common.c(173): warning C4996: 'PySys_SetArgvEx': deprecated in 3.11
      "C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\bin\HostX86\x64\link.exe" /nologo /INCREMENTAL:NO /LTCG /MANIFEST:EMBED,ID=1 /LIBPATH:C:\Users\Kenneth.Miller\Work\python\mcm\venv\libs /LIBPATH:C:\Users\Kenneth.Miller\AppData\Local\Programs\Python\Python311\libs /LIBPATH:C:\Users\Kenneth.Miller\AppData\Local\Programs\Python\Python311 /LIBPATH:C:\Users\Kenneth.Miller\Work\python\mcm\venv\PCbuild\amd64 "/LIBPATH:C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\ATLMFC\lib\x64" "/LIBPATH:C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\lib\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\lib\um\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.22000.0\ucrt\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\\lib\10.0.22000.0\\um\x64" user32.lib build\temp.win-amd64-cpython-311\Release\source/bases/Win32GUI.obj /OUT:build\lib.win-amd64-cpython-311\cx_Freeze\bases\Win32GUI-cp311-win_amd64.exe /MANIFEST
      Generating code
      Finished generating code
      "C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\bin\HostX86\x64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -Isource/bases -IC:\Users\Kenneth.Miller\Work\python\mcm\venv\include -IC:\Users\Kenneth.Miller\AppData\Local\Programs\Python\Python311\include -IC:\Users\Kenneth.Miller\AppData\Local\Programs\Python\Python311\Include "-IC:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include" "-IC:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\ATLMFC\include" "-IC:\Program Files\Microsoft Visual Studio\2022\Community\VC\Auxiliary\VS\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\um" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\shared" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\winrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\cppwinrt" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um" /Tcsource/bases/Win32Service.c /Fobuild\temp.win-amd64-cpython-311\Release\source/bases/Win32Service.obj
      Win32Service.c
      C:\Users\Kenneth.Miller\AppData\Local\Temp\pip-install-nd7imgba\cx-freeze_fb866067fc8f42a9bec23e8ae76289f0\source\bases\common.c(170): warning C4996: 'Py_SetProgramName': deprecated in 3.11
      C:\Users\Kenneth.Miller\AppData\Local\Temp\pip-install-nd7imgba\cx-freeze_fb866067fc8f42a9bec23e8ae76289f0\source\bases\common.c(171): warning C4996: 'Py_SetPath': deprecated in 3.11
      C:\Users\Kenneth.Miller\AppData\Local\Temp\pip-install-nd7imgba\cx-freeze_fb866067fc8f42a9bec23e8ae76289f0\source\bases\common.c(173): warning C4996: 'PySys_SetArgvEx': deprecated in 3.11
      "C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\bin\HostX86\x64\link.exe" /nologo /INCREMENTAL:NO /LTCG /MANIFEST:EMBED,ID=1 /LIBPATH:C:\Users\Kenneth.Miller\Work\python\mcm\venv\libs /LIBPATH:C:\Users\Kenneth.Miller\AppData\Local\Programs\Python\Python311\libs /LIBPATH:C:\Users\Kenneth.Miller\AppData\Local\Programs\Python\Python311 /LIBPATH:C:\Users\Kenneth.Miller\Work\python\mcm\venv\PCbuild\amd64 "/LIBPATH:C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\ATLMFC\lib\x64" "/LIBPATH:C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\lib\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\lib\um\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.22000.0\ucrt\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\\lib\10.0.22000.0\\um\x64" advapi32.lib cx_Logging.lib delayimp.lib build\temp.win-amd64-cpython-311\Release\source/bases/Win32Service.obj /OUT:build\lib.win-amd64-cpython-311\cx_Freeze\bases\Win32Service-cp311-win_amd64.exe /DELAYLOAD:cx_Logging.cp311-win_amd64.pyd /MANIFEST
      LINK : fatal error LNK1181: cannot open input file 'cx_Logging.lib'
      error: command 'C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Tools\\MSVC\\14.34.31933\\bin\\HostX86\\x64\\link.exe' failed with exit code 1181
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for cx_Freeze
Failed to build cx_Freeze
ERROR: Could not build wheels for cx_Freeze, which is required to install pyproject.toml-based projects

**Comment by foreachthing** [comment]
I get the same thing on Windows 10 and Python 3.11.0:
```
LINK : fatal error LNK1181: cannot open input file 'cx_Logging.lib'
      error: command 'C:\\Program Files\\Microsoft Visual Studio\\2022\\Professional\\VC\\Tools\\MSVC\\14.34.31933\\bin\\HostX86\\x64\\link.exe' failed with exit code 1181
      [end of output]
```

**Reply by foreachthing** [reply]
Just found this here https://github.com/marcelotduarte/cx_Freeze/discussions/1394#discussioncomment-2269475 and it helped with my installation. Thanks!

**Comment by marcelotduarte** [comment, marked as answer]
cx_Freeze supports Python 3.11 in dev version for now. Use:
`pip install --upgrade --pre --extra-index-url https://marcelotduarte.github.io/packages/ cx_Freeze`

**Reply by foreachthing** [reply]
Thanks, Marcelo!!

```
Successfully installed cx_Freeze-6.14.0.dev1
```

**Reply by foreachthing** [reply]
Howevery, I still had to install `cx_Logging` first. Only then the installation of cx_freeze worked.
